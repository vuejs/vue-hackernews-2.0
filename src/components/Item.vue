<template>
  <li class="news-item">
    <div class="score"></div>
    <div class="title">
      <template v-if="item.url">
        <a :href="item.url" target="_blank" rel="noopener">{{ item.title }}</a>
        <span class="host">&nbsp;({{ item.url | host }})</span>
      </template>
      <template v-else>
        <router-link :to="'/item/' + item.id">{{ item.title }}</router-link>
      </template>
    </div>
    <span class="meta">
      <span v-if="item.type !== 'job'" class="by">
        by
        <router-link :to="'/user/' + item.by">{{ item.by }}</router-link>
      </span>
      <span class="time">&nbsp;{{ item.time | timeAgo }} ago</span>
      <span v-if="item.type !== 'job'" class="comments-link">
        |
        <router-link :to="'/item/' + item.id">{{ item.descendants }} comments</router-link>
      </span>

      <span>&nbsp;| {{item.score}} points</span>
    </span>
    <br />
    <br />
    <similar v-if="item.type === 'story'" :story="item"></similar>
    <span class="label" v-if="item.type !== 'story'">{{ item.type }}</span>
  </li>
</template>

<script>
import { timeAgo } from "../util/filters";
import Similar from "./Similar.vue";

export default {
  name: "news-item",
  components: { Similar },
  props: ["item"],
  // http://ssr.vuejs.org/en/caching.html#component-level-caching
  serverCacheKey: ({ item: { id, __lastUpdated, time } }) => {
    return `${id}::${__lastUpdated}::${timeAgo(time)}`;
  }
};
</script>

<style lang="stylus">
.news-item {
  background-color: #FFFEF2;
  padding: 20px 30px 20px 30px;
  border-bottom: 1px solid #eee;
  position: relative;
  line-height: 20px;

  .meta, .host {
    font-size: 0.85em;
    color: #828282;

    a {
      color: #828282;
      text-decoration: underline;

      &:hover {
        color: #F7955B;
      }
    }
  }
}
</style>
