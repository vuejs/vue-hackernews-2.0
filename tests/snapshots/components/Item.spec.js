import { mountWithPlugins } from '../../unit/test-utils/mountWithPlugins';
import Item from '@/components/Item.vue';

describe('snapshot - Item.vue', () => {
    // test that the item is rendering in the same way as before
    it('renders correctly', () => {
        // we create a wrapper and pass some mock data as the prop "item"
        const wrapper = mountWithPlugins(Item, {
            propsData: {
                item: {
                    id: 1,
                    score: 22,
                    url: 'https://www.google.com/',
                    type: 'job'
                }
            }
        });
        expect(wrapper.element).toMatchSnapshot();
    })
});
