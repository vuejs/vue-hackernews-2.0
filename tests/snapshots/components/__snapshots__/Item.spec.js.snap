// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot - Item.vue renders correctly 1`] = `
<li
  class="news-item"
>
  <span
    class="score"
  >
    22
  </span>
   
  <span
    class="title"
  >
    <a
      href="https://www.google.com/"
      rel="noopener"
      target="_blank"
    >
      
    </a>
     
    <span
      class="host"
    >
       (google.com)
    </span>
  </span>
   
  <br />
   
  <span
    class="meta"
  >
    <!---->
     
    <span
      class="time"
    >
      
      0 days ago
    
    </span>
     
    <!---->
  </span>
   
  <span
    class="label"
  >
    job
  </span>
</li>
`;
